src/dev/impl/DevToys/Views/Tools/Generators/HashGenerator/HashGeneratorToolPage.xaml

```diff
@@ -50,6 +61,13 @@
             AcceptsReturn="True"
             Text="{x:Bind ViewModel.Input, Mode=TwoWay, UpdateSourceTrigger=PropertyChanged}"/>
 
+        <controls:CustomTextBox x:Name="SecretKeyInput"
+            Header="Secret Key (HMAC only)"
```

Comment:
Same thing here. Also, perhaps we can remove `(HMAC only)` since this field only appear when `HMAC mode` is enabled.